rif-collector-app.php

```diff
@@ -34,4 +34,92 @@ function retrieveAllQuery($db) {
+    }
+    if (!is_string($entryToAdd['color'])) {
+        return $message = "Error with your 'Color' input";
+    }
+    if (!filter_var($entryToAdd['mags'], FILTER_VALIDATE_INT)) {
+        return $message = "Error with your 'Mags' input";
+    }
+    if (!is_string($entryToAdd['power'])) {
+        return $message = "Error with your 'Power' input";
+    }
+    if (!filter_var($entryToAdd['sites'], FILTER_VALIDATE_INT)) {
+        return $message = "Error with your 'Sites Visited' input";
+    }
+    if (!preg_match($dateRegex, $entryToAdd['purchased'])) {
+        return $message = "Error with your 'Purchased Date' input";
+    } else {
+
+        $color = strip_tags($entryToAdd['color']);
+        $color = strtolower($color);
+
+        if ($color === 'white') {
+            $color = 3;
+        } elseif ($entryToAdd['color'] === 'black') {
+            $color = 2;
+        } elseif ($entryToAdd['color'] === 'tan') {
+            $color = 1;
+        } else {
+            return $message = "Error with your 'Color' input";
+        }
+
+    return $validatedSanitizedArr = [
+        'make' => htmlspecialchars($entryToAdd['make']),
+        'model' => htmlspecialchars($entryToAdd['model']),
+        'type' => htmlspecialchars($entryToAdd['type']),
+        'color' => $color,
+        'mags' => filter_var($entryToAdd['mags'], FILTER_SANITIZE_NUMBER_INT),
+        'power' => htmlspecialchars($entryToAdd['power']),
+        'sites' => filter_var($entryToAdd['sites'], FILTER_SANITIZE_NUMBER_INT),
+        'purchased' => ($entryToAdd['purchased']),
+        ];
+    }
+}
+
+function addToDatabase(array $valSanArr, object $db, string &$message) {
+
+    $addQuery = $db->prepare("INSERT INTO `rifs` (`make`, `model`, `type`, `color`, `mags_owned`, `power_source`, `sites_visited`, `purchase_date`) VALUES (:make, :model, :type, :color, :mags, :power, :sites, :purchased) ");
+    $result = $addQuery->execute([
+        'make' => $valSanArr['make'],
+        'model' => $valSanArr['model'],
+        'type' => $valSanArr['type'],
+        'color' => $valSanArr['color'],
+        'mags' => $valSanArr['mags'],
+        'power' => $valSanArr['power'],
+        'sites' => $valSanArr['sites'],
+        'purchased' => $valSanArr['purchased']
+    ]);
+
+    if (!$result) {
+        return $message = "There has been an error in adding to database";
+    } else {
+        return $message = 'Entry Added successfully';
+    }
+}
+
+function hasFormBeenSubmitted(array $array, $validatedSanitizedArr, $db, &$message) {
+    if ( (isset($array['make'])) && (isset($array['model'])) && (isset($array['type'])) && (isset($array['color'])) && (isset($array['mags'])) && (isset($array['power'])) && (isset($array['sites'])) && (isset($array['purchased'])) ) {
+        validateSanitizeEntry($array, $validatedSanitizedArr, $message);
+        if ($message === '') {
+            addToDatabase($validatedSanitizedArr, $db, $message);
+        }
+    }
```

Comment:
`hasFormBeenSubmitted` is kind of a redundant function - it just calls other functions and you pass in the `$_POST`... superglobals never need to be passed in, so you actually disguise what you're doing by bringing this into a function and calling the `$_POST` the very un-specific 'array'.

These lines could just go in the index.php and that would be fine - this is code that that page needs to do to build that page.

Reply:
Actioned